linalg/sparsemat.hpp

```diff
@@ -216,7 +216,7 @@ class SparseMatrix : public AbstractSparseMatrix
    void ClearCuSparse() { ClearGPUSparse(); }
 
    /// Check if the SparseMatrix is empty.
-   bool Empty() const { return (A == NULL) && (Rows == NULL); }
+   bool Empty() const { return A.Empty() && (Rows == NULL); }
```

Comment:
Maybe this change can be made in the other places with `A == NULL` comparison? I ran into the same issue in `SparseMatrix::PrintMatlab`.

Reply:
Sure, feel free to add any other issues you noticed.